cas/lib/cas/impl.ex

```diff
@@ -0,0 +1,36 @@
+defmodule CAS.Impl do
+
+    def start() do
+        get_user_type()
+    end
+
+    #Get user type from input. Input 1 stands for Doctors. Input 2 stands for Patients.
+    defp get_user_type() do
+        user_type = IO.gets "Please Enter User Type: 1 for Doctors; 2 for Patients." |> String.trim
+        input_validation(user_type)
+    end
+
+    #Validate user type input. "1" or "2" only.
```

Comment:
I would structure this differently:

~~~
defp get_user_type() do
  IO.gets ...
  |> String.trim
  |> select_based_on_type()
end

defp select_based_on_type("1") do
  # doctor stuff
end

defp select_based_on_type("2) do
 # patient stuff
end

defp select_based_on_type(other) do
  IO.puts "invalid option #{other)"
  get_user_type()
end
~~~